src/PaymentGateways/Gateways/PayPalCommerce/payPalCommerceGateway.tsx

```diff
@@ -181,18 +170,50 @@ import createSubscriptionPlan from './resources/js/createSubscriptionPlan';
         return formData;
     };
 
-    const validateHostedFields = () => {
-        return Object.values(hostedField.cardFields.getState().fields).some(
-            (field: {isValid: boolean}) => field.isValid
+    const smartButtonsCreateOrderHandler: PayPalButtonsComponentOptions['createOrder'] = async (): Promise<string> => {
+        return await createOrder(
+            `${payPalDonationsSettings.ajaxUrl}?action=give_paypal_commerce_create_order`,
+            payPalCommerceGateway,
+            getFormData()
+        );
+    };
+
+    /**
+     * @unreleased
+     */
+    const cardFieldsOnErrorHandler: PayPalCardFieldsComponentBasics['onError'] = (error) => {
+        console.error('PayPal Card Fields Error:', error);
+        // Handle the error as needed
```

Comment:
Should we throw an exception here to provide some user feedback?

Reply:
Sure we can try, I might need to set a form level using setError but i'll test that out.

Reply:
@glaubersilva it turns out we have to handle errors on the server so I just updated the gateway to handle the ones from their docs https://github.com/impress-org/givewp/pull/7850/commits/d9e41a83a312855019a43fb72cba487d41363546

Reply:
Here's some additional error cards to test out https://developer.paypal.com/tools/sandbox/card-testing/#link-simulatecarderrorscenarios